pam/pam_test/module-transaction-dummy.go

```diff
@@ -0,0 +1,346 @@
+	pam.StringConvResponse, error) {
+	if style == pam.BinaryPrompt {
+		return nil, pam.NewTransactionError(errors.New("Binary style is not supported"),
+			pam.ConvErr)
+	}
+
+	res, err := m.StartConv(pam.NewStringConvRequest(style, prompt))
+	if err != nil {
+		return nil, err
+	}
+
+	stringRes, ok := res.(pam.StringConvResponse)
+	if !ok {
+		return nil, pam.NewTransactionError(errors.New("Can't convert to pam.StringConvResponse"), pam.ConvErr)
+	}
+	return stringRes, nil
+}
+
+// StartStringConvf allows to start string conversation with formatting support.
+func (m *ModuleTransactionDummy) StartStringConvf(style pam.Style, format string, args ...interface{}) (
+	pam.StringConvResponse, error) {
+	return m.StartStringConv(style, fmt.Sprintf(format, args...))
+}
+
+// StartBinaryConv starts a binary conversation using the provided bytes.
+func (m *ModuleTransactionDummy) StartBinaryConv(bytes []byte) (
+	pam.BinaryConvResponse, error) {
+	res, err := m.StartConv(NewBinaryRequestDummyFromBytes(bytes))
+	if err != nil {
+		return nil, err
+	}
+
+	binaryRes, ok := res.(pam.BinaryConvResponse)
+	if !ok {
+		return nil, pam.NewTransactionError(
+			errors.New("Can't convert to pam.BinaryConvResponse"), pam.ConvErr)
+	}
+	return binaryRes, nil
+}
+
+// StartConv initiates a PAM conversation using the provided ConvRequest.
+func (m *ModuleTransactionDummy) StartConv(req pam.ConvRequest) (
+	pam.ConvResponse, error) {
+	resp, err := m.StartConvMulti([]pam.ConvRequest{req})
+	if err != nil {
+		return nil, err
+	}
+	if len(resp) != 1 {
+		return nil, pam.NewTransactionError(errors.New("not enough values returned"),
+			pam.ConvErr)
+	}
+	return resp[0], nil
+}
+
+// StartConvMulti initiates a PAM conversation with multiple ConvRequest's.
+func (m *ModuleTransactionDummy) StartConvMulti(requests []pam.ConvRequest) (
+	[]pam.ConvResponse, error) {
+	if len(requests) == 0 {
+		return nil, pam.NewTransactionError(errors.New("no requests defined"),
+			pam.ConvErr)
+	}
+
+	goReplies := make([]pam.ConvResponse, 0, len(requests))
+	for _, req := range requests {
+		msgStyle := req.Style()
+		switch msgStyle {
+		case pam.PromptEchoOff:
+			fallthrough
+		case pam.PromptEchoOn:
+			fallthrough
+		case pam.ErrorMsg:
+			fallthrough
+		case pam.TextInfo:
+			if m.convHandler == nil {
+				return nil, pam.NewTransactionError(
+					fmt.Errorf(
+						"no conversation handler provided for style %v", msgStyle),
+					pam.ConvErr)
+			}
+			reply, err := m.convHandler.RespondPAM(msgStyle,
+				req.(pam.StringConvRequest).Prompt())
+			if err != nil {
+				return nil, err
+			}
+			goReplies = append(goReplies, StringResponseDummy{
+				msgStyle,
+				reply,
+			})
+		case pam.BinaryPrompt:
+			handler, ok := m.convHandler.(pam.BinaryConversationHandler)
+			if handler == nil || !ok {
+				return nil, pam.NewTransactionError(
+					errors.New("no binary handler provided"),
+					pam.ConvErr)
+			}
+			binReq, _ := req.(pam.BinaryConvRequester)
```

Comment:
Honestly, here... I would just have used `binReq := req.(pam.BinaryConvRequester)` to panic if this is not true.

By doing this we'd instead panic because of nil pointer deference, that is still something fine.

I mean, it's a programmer error if this is not true, so I would like just panic instead of doing an error check.